routemaster/services/metrics/datadog_adapter.rb

```diff
@@ -3,23 +3,36 @@
 
 module Routemaster
   module Services
-    module MetricsCollectors
-      class Datadog
+    module Metrics
+      class DatadogAdapter
         include Singleton
+        include MonitorMixin
 
         def initialize
           require 'dogapi'
           api_key = ENV.fetch('DATADOG_API_KEY')
           app_key = ENV.fetch('DATADOG_APP_KEY')
           @dog ||= Dogapi::Client.new(api_key, app_key)
+          super
         rescue KeyError
           abort 'Please install and configure datadog (or equivalent service) first!'
         end
 
-        def perform(name, value, tags = [])
-          @dog.emit_point(name, value, tags: tags)
+        def batched
+          synchronize do
+            @dog.batch_metrics do
+              yield
+            end
+          end
         end
 
+        def gauge(name, value, tags)
+          synchronize do
+            @dog.emit_point("routemaster.#{name}", value, tags: tags, type: __callee__.to_s)
```

Comment:
The use of `__callee__` here seems rather non-obvious, relying on people knowing that it depends on method aliasing. It took me a minute to work out what was going on. While it might be a bit longer I'd probably make this into a private method `emit_point` which takes a type and then have specific `gauge` and `counter` methods.